openzwave-sys/src/value_classes/value_id.rs

```diff
@@ -24,17 +24,44 @@ pub enum ValueType {
     //ValueType_Max = ValueType_Raw // likely useless in Rust wrapper
 }
 
-pub enum ValueID {}
+use std::fmt;
+impl fmt::Display for ValueType {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.pad(match self {
+            &ValueType::ValueType_Bool => "Bool",
+            &ValueType::ValueType_Byte => "Byte",
+            &ValueType::ValueType_Decimal => "Decimal",
+            &ValueType::ValueType_Int => "Int",
+            &ValueType::ValueType_List => "List",
+            &ValueType::ValueType_Schedule => "Schedule",
+            &ValueType::ValueType_Short => "Short",
+            &ValueType::ValueType_String => "String",
+            &ValueType::ValueType_Button => "Button",
+            &ValueType::ValueType_Raw => "Raw",
+        })
+    }
+}
+
+
+
+// IMPORTANT: This ValueID struct MUST match the layout of the OpenZWave
+//            ValueID class.
+#[repr(C)]
+pub struct ValueID {
+  pub id: u32,
+  pub id1: u32,
+  pub home_id: u32,
+}
```

Comment:
I don't think that we need to from the internal ValueID (the above ValueID is the rust representation of the C++ ValueID class).
We already derive Clone and Hash for the rust ValueID